controlplane/kubeadm/controllers/controller.go

```diff
@@ -223,6 +212,33 @@ func (r *KubeadmControlPlaneReconciler) Reconcile(req ctrl.Request) (res ctrl.Re
 	return r.reconcile(ctx, cluster, kcp)
 }
 
+func patchKubeadmControlPlane(ctx context.Context, patchHelper *patch.Helper, kcp *controlplanev1.KubeadmControlPlane) error {
+	// Always update the readyCondition by summarizing the state of other conditions.
+	conditions.SetSummary(kcp,
+		conditions.WithConditions(
+			controlplanev1.MachinesSpecUpToDateCondition,
+			controlplanev1.ResizedCondition,
+			controlplanev1.MachinesReadyCondition,
+			controlplanev1.AvailableCondition,
+			controlplanev1.CertificatesAvailableCondition,
+		),
+	)
+
+	// Patch the object, ignoring conflicts on the conditions owned by this controller.
+	return patchHelper.Patch(
+		ctx,
+		kcp,
+		patch.WithOwnedConditions{Conditions: []clusterv1.ConditionType{
+			clusterv1.ReadyCondition,
+			controlplanev1.MachinesSpecUpToDateCondition,
+			controlplanev1.ResizedCondition,
+			controlplanev1.MachinesReadyCondition,
+			controlplanev1.AvailableCondition,
+			controlplanev1.CertificatesAvailableCondition,
+		}},
+	)
```

Comment:
Seems there is a lot of repetition here, is there a way we could have a `conditions.Owned` method when doing the SetSummary

Reply:
There is a repetition between Summary and Owned conditions, but they are not the same.
However I agree there is a pattern that repeats across controllers, so this is probably something we can address within the context of https://github.com/kubernetes-sigs/cluster-api/issues/3517

Reply:
Why do we need owned conditions here? I guess this is actually going to be solved once we get server side apply in place

Reply:
Owned conditions are required to avoid reconcile conflicts that can happen due to stale reads or to patch issued in the middle of a reconcile loop.
With server-side apply some of these problems a going away, but I'm not sure all of them.

Reply:
Sounds good, let's revisit later, I'm not 100% sure this is the direction we want to go long term, but seems fine for now